### PR TITLE
Feature/verfy seal

### PR DESCRIPTION
**Pull request by artyom-yurin**
### Description of the Change
Seal functions:
- VerifySeal
- SealCommit
- Unseal
- UnsealRange

### Benefits
Using Seal

### Possible Drawbacks 
None

### CODE REVIEW

File: test/core/proofs/proofs_test.cpp [hunk 1]
```diff
@@ -185,3 +185,237 @@ TEST_F(ProofsTest, DISABLED_ValidPoSt) {
...
+    ch = file_c.get();
+    while (!file_c.eof()) {
+        file_c_bytes.push_back(ch);
+        ch = file_c.get();
+    }
+
+    ASSERT_EQ(file_c_bytes.size(), 508);
+    ASSERT_EQ(gsl::span<uint8_t>(file_c_bytes.data(), 508),
+              gsl::span<uint8_t>(some_bytes.data(), 508));
+}
```
Comment (Alexey-N-Chernyshov) [inline test/core/proofs/proofs_test.cpp]: EOF

**Comment by Alexey-N-Chernyshov** [comment]
💯

File: core/proofs/proofs.cpp [hunk 2]
```diff
@@ -444,13 +547,47 @@ namespace fc::proofs {
...
+      const std::string &sealed_sector_path,
+      const std::string &unseal_output_path,
+      uint64_t sector_id,
+      const Prover &prover_id,
+      const Ticket &ticket,
+      const Comm &comm_d,
+      uint64_t offset,
+      uint64_t length) {
+    auto res_ptr =
+        unseal_range(cSectorClass(sector_size, porep_proof_partitions),
```
Comment (turuslan) [inline core/proofs/proofs.cpp]: make_unique

File: test/core/proofs/proofs_test.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply test/core/proofs/proofs_test.cpp]: Settings → Editor → General → Ensure line feed at file end on Save
:suspect:

**Pull request merged**